packages/kbn-unified-field-list/src/hooks/use_grouped_fields.ts

```diff
@@ -120,13 +126,34 @@ export function useGroupedFields<T extends FieldListItem = DataViewField>({
     };
 
     const selectedFields = sortedSelectedFields || [];
-    const sortedFields = [...(allFields || [])].sort(sortFields);
+
+    // Taking care of new fields that were ingested after the selected data view was loaded
+    // Those replace existing fields if updated, or are added to the list
+    const newFields =
+      dataViewId && getNewFieldsBySpec
+        ? getNewFieldsBySpec(fieldsExistenceReader.getNewFields(dataViewId), dataView)
+        : [];
+    // Filtering out fields that e.g. Discover provides by analyzing the resultset, that are not part of the loaded DataView
+    // These can be replaced by the new fields, which are mapped correctly, and therefore can be used in the right way
+    const allFieldsExlNew =
+      allFields && newFields.length
+        ? allFields.filter((field) => !newFields.some((newField) => newField.name === field.name))
+        : allFields;
+
+    const allFieldsInclNew = newFields.length
+      ? [...(allFieldsExlNew || []), ...newFields]
+      : allFields || [];
+    allFieldsToReturn.current = newFields.length ? allFieldsInclNew : allFields;
+    hasNewFields.current = Boolean(newFields.length);
+    const sortedFields = [...allFieldsInclNew].sort(sortFields);
+
```

Comment:
How about we extract this logic into a separate hook and wrap it in a memo which recalculates only if `allFields` or `newFields` changes? It does not look like it has to be inside `useGroupedFields` hook.

Reply:
yes, the makes sense, it's simplifying this logic a log, I pushed this new hook, waiting for CI, interested if this matches your expectations?

Reply:
here's the latest commit: 26c4d72c10ff4f4f3a70209a679625eba20d2bec